parser/src/test/java/com/facebook/coresql/parser/sqllogictest/java/SqlLogicTest.java

```diff
@@ -0,0 +1,336 @@
+import static java.lang.Math.min;
+import static java.nio.file.FileVisitResult.CONTINUE;
+import static java.nio.file.FileVisitResult.TERMINATE;
+import static java.nio.file.Files.createFile;
+import static java.nio.file.Files.isHidden;
+import static java.nio.file.Files.readAllLines;
+import static java.nio.file.Files.walkFileTree;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+public final class SqlLogicTest
+{
+    private static final Logger LOG = Logger.get(SqlLogicTest.class);
+    private final String nameOfResultFile;
+    private final SqlLogicTestResult result;
+    private final FileVisitor<Path> fileVisitor;
+    private final Path testFilesRootPath;
+    private final Path resultPath;
+    private final DatabaseDialect databaseDialect;
+    private final int maxNumFilesToVisit;
+
+    enum CoreSqlParsingError
+    {
+        PARSING_ERROR,
+        UNPARSING_ERROR,
+        UNPARSED_DOES_NOT_MATCH_ORIGINAL_ERROR
+    }
+
+    enum DatabaseDialect
+    {
+        ALL,
+        ORACLE,
+        MYSQL,
+        MSSQL,
+        SQLITE,
+        POSTGRESQL
+    }
+
+    @Inject
+    public SqlLogicTest(SqlLogicTestConfig config)
+    {
+        this.nameOfResultFile = requireNonNull(config.getNameOfResultFile(), "name of result file is null");
+        this.testFilesRootPath = requireNonNull(config.getTestFilesRootPath(), "test files root path is null");
+        this.resultPath = requireNonNull(config.getResultPath(), "result path is null");
+        this.databaseDialect = requireNonNull(config.getDatabaseDialect(), "database dialect is null");
+        this.result = new SqlLogicTestResult(config.getDatabaseDialect());
+        this.fileVisitor = new SqlLogicTestFileVisitor();
+        this.maxNumFilesToVisit = config.getMaxNumFilesToVisit();
+    }
+
+    public static void execute()
+            throws IOException
+    {
+        Bootstrap app = new Bootstrap(new SqlLogicTestModule());
+
+        Injector injector = app
+                .doNotInitializeLogging()
+                .quiet()
+                .initialize();
+
+        injector.getInstance(SqlLogicTest.class).run();
+    }
+
+    private void run()
+            throws IOException
+    {
+        createFileForResult(resultPath, nameOfResultFile);
+        walkFileTree(testFilesRootPath, fileVisitor);
+        writeTestResultToFile(result, resultPath, nameOfResultFile);
+    }
+
+    private static void createFileForResult(Path resultPath, String nameOfResultFile)
+            throws IOException
+    {
+        try {
+            createFile(resultPath.resolve(nameOfResultFile));
+        }
+        catch (FileAlreadyExistsException e) {
+            LOG.info("Result file already exists. We'll overwrite it with the new result.");
+        }
+        catch (IOException e) {
+            throw new IOException("Could not create the sqllogictest result file.");
```

Comment:
Don't swallow the exception here, because it will be difficult to debug. Pass the throwable to the constructor, then we can keep the exception chain.

Reply:
Done.